### PR TITLE
MySQL: Add CREATE DATABASE and ALTER DATABASE

### LINKED ISSUE #4259: Unparseable section on CREATE DATABASE with DEFAULT CHARACTER SET
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Parser raises "Found unparsable section" error on the following:
```sql
CREATE DATABASE IF NOT EXISTS 'db_name' DEFAULT CHARACTER SET utf8;
```

The same thing happens with this example, taken from the [MariaDB documentation](https://mariadb.com/kb/en/create-database/):
```sql
CREATE DATABASE czech_slovak_names 
  CHARACTER SET = 'keybcs2'
  COLLATE = 'keybcs2_bin';
```

### Expected Behaviour

Parser should parse the section correctly.

### Observed Behaviour

See above.

### How to reproduce

1. Save the code given above in an `.sql` file
2. Specify the "mysql" dialect either in `.sqlfluff`, or as a commandline parameter
3. Run `sqlfluff lint filename.sql`

### Dialect

mysql

### Version

1.4.5

### Configuration

```ini
[sqlfluff]
dialect = mysql
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by yoichi**
### Brief summary of the change made

fixes #4259

### Are there any other side effects of this change that we should be aware of?
No

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: test/fixtures/dialects/mysql/create_database.sql [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+DEFAULT CHARACTER SET utf8mb4
+COLLATE utf8mb4_0900_ai_ci
+DEFAULT ENCRYPTION 'N';
+
+CREATE DATABASE my_database
+DEFAULT CHARACTER SET = utf8mb4
+COLLATE = utf8mb4_0900_ai_ci
+DEFAULT ENCRYPTION = 'N';
+
+CREATE SCHEMA my_database
```
Comment (WittierDinosaur) [inline test/fixtures/dialects/mysql/create_database.sql:15]: Can you separate this out into a different file? It's not the same statement type

File: test/fixtures/dialects/mysql/alter_database.sql [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+DEFAULT CHARACTER SET utf8mb4
+COLLATE utf8mb4_0900_ai_ci
+DEFAULT ENCRYPTION 'N';
+
+ALTER DATABASE my_database
+DEFAULT CHARACTER SET = utf8mb4
+COLLATE = utf8mb4_0900_ai_ci
+DEFAULT ENCRYPTION = 'N';
+
+ALTER SCHEMA my_database
```
Comment (WittierDinosaur) [inline test/fixtures/dialects/mysql/alter_database.sql:11]: Same here - also, not sure if intended, but you haven't changed the alter schema grammar at all

**Review by WittierDinosaur — CHANGES_REQUESTED** [review]

File: test/fixtures/dialects/mysql/create_database.sql (hunk 1 above)
Comment (yoichi) [reply test/fixtures/dialects/mysql/create_database.sql:15]: On MySQL, CREATE SCHEMA is a synonym for CREATE DATABASE, as described in https://dev.mysql.com/doc/refman/8.0/en/create-database.html

So, I think it's better to combine statements than separate tests. What do you think?
However, I don't know how to disable ansi.CreateSchemaStatementSegment.
If you know how to do it, please let me know.

File: test/fixtures/dialects/mysql/alter_database.sql (hunk 2 above)
Comment (yoichi) [reply test/fixtures/dialects/mysql/alter_database.sql:11]: On MySQL, ALTER SCHEMA is a synonym for ALTER DATABASE, as described in https://dev.mysql.com/doc/refman/8.0/en/alter-database.html.

So, I've implemented it in AlterDatabaseStatementSegment.

File: test/fixtures/dialects/mysql/create_database.sql (hunk 1 above)
Comment (yoichi) [reply test/fixtures/dialects/mysql/create_database.sql:15]: > However, I don't know how to disable ansi.CreateSchemaStatementSegment.
> If you know how to do it, please let me know.

I found 'remove' argument of copy() is for that. I'll try.

**Review by WittierDinosaur — APPROVED** [review]
LGTM

**Pull request merged**